src/client/Client.cc

```diff
@@ -10577,12 +10577,13 @@ int64_t Client::_write(Fh *f, int64_t offset, uint64_t size, const char *buf,
   ceph_assert(ceph_mutex_is_locked_by_me(client_lock));
 
   uint64_t fpos = 0;
+  Inode *in = f->inode.get();
 
-  if ((uint64_t)(offset+size) > mdsmap->get_max_filesize()) //too large!
-    return -CEPHFS_EFBIG;
-
+  if ( (uint64_t)(offset+size) > mdsmap->get_max_filesize() && //exceeds config
+       (uint64_t)(offset+size) > in->size ) { //exceeds filesize 
+      return -CEPHFS_EFBIG;              
```

Comment:
To be more readable maybe we should do:

`(uint64_t)(offset+size) > std::max(in->size, mdsmap->get_max_filesize())`

Please add a comment to explain why the `in->size` will larger than `mdsmap->get_max_filesize()`

Reply:
I agree to this.

Reply:
I agree it is more readable but also want to point that extra comparison(std::max) will be always executed for this path while the case of `in->size > mdsmap->get_max_filesize()` is quit rare.

I"ll soon push the change and add a comment.